tests/integration-test/src/main.rs

```diff
@@ -8,10 +11,33 @@ fn main() -> anyhow::Result<()> {
     builder.filter_level(LevelFilter::Debug);
     builder.init();
 
+    let args: Vec<String> = env::args().collect();
+    // Ignore the first arg, which is the function name
+    let tests_to_run = &args[1..args.len()];
+    let tests_to_run_len = tests_to_run.len();
+
+    if tests_to_run_len > 0 {
+        info!("Executing test case(s): {:?}", tests_to_run);
+    } else {
+        info!("Executing all test cases");
+    }
+
+    // Get the tests into a collection that can be sorted.
+    let mut tests: Vec<<iter<IntegrationTest> as IntoIterator>::Item> =
+        inventory::iter::<IntegrationTest>.into_iter().collect();
+
+    // Sort the tests so we can excute them in the desired order
+    tests.sort();
+
     // Run the tests
-    for t in inventory::iter::<IntegrationTest> {
-        info!("Running {}", t.name);
-        (t.test_fn)();
+    for t in tests {
+        let test_name: Vec<&str> = t.name.split("::").collect();
+        let test_name = test_name[test_name.len() - 1].to_string();
+
```

Comment:
```suggestion
    let test_name: String = t.name.split("::").collect::<Vec<&str>>().pop().expect("not a valid test name").to_string();


```

Reply:
Much more rusty.  I couldn't figure out how to do it in one line.